### PR TITLE
fix: Expose Taskoptions for SubmitTask* type in Extensions.Csharp

### PR DESCRIPTION
**Pull request by ddubuc**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -206,7 +206,7 @@ jobs:
 
       - id: deploy
         name: Deploy
-        uses: aneoconsulting/ArmoniK.Action.Deploy/deploy@main
+        uses: aneoconsulting/ArmoniK.Action.Deploy/deploy@dd/fixUploadArtifact
```
Comment (lemaitre-aneo) [inline .github/workflows/build.yml:209]: Revert back to main. The PR on Action.Deploy has been merged (https://github.com/aneoconsulting/Armonik.Action.Deploy/pull/7)

File: Client/src/Unified/Services/Submitter/Service.cs [hunk 2]
```diff
@@ -143,10 +141,31 @@ public class Service : AbstractClientService, ISubmitterService
...
+                                    throw new
+                                      InvalidOperationException("The taskOptions maxRetries in AsyncBuffer has at least one different value. Please bundle tasks with same maxRetries");
+                                  }
+
+                                  if (distinctPriority > 1)
+                                  {
+                                    throw new
+                                      InvalidOperationException("The taskOptions priority in AsyncBuffer has at least one different value. Please bundle tasks with same Priority");
+                                  }
+
```
Comment (lemaitre-aneo) [inline Client/src/Unified/Services/Submitter/Service.cs:162]: What does this has to do with TaskOptions?

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/LargeSubmitAsync/LargeSubmitAsyncClient.cs [hunk 3]
```diff
@@ -224,18 +225,21 @@ public override void EntryPoint()
...
-                             resultTask.Add(task.Result);
+                             resultTask.Add(task);
                            }
                          });
 
-    return resultTask.ToList();
+    resultTask.AsParallel()
+              .ForAll(task => results.Add(task.Result));
+
+    return results.ToList();
```
Comment (lemaitre-aneo) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/LargeSubmitAsync/LargeSubmitAsyncClient.cs:242]: What do this has to do with TaskOptions?
Besides, you could most likely use `.Select` instead of `.ForEach` that adds into a concurrentbag

**Review by lemaitre-aneo — COMMENTED** [review]
What remains to be done on this one?

File: Client/src/Unified/Services/Submitter/Service.cs (hunk 2 above)
Comment (aneojgurhem) [reply Client/src/Unified/Services/Submitter/Service.cs:162]: Can't do a groupby ?

File: .github/workflows/build.yml (hunk 1 above)
Comment (aneojgurhem) [reply .github/workflows/build.yml:209]: @dbrasseur-aneo can you do it ?

File: Client/src/Common/Submitter/BaseClientSubmitter.cs [hunk 4]
```diff
@@ -529,12 +543,21 @@ public ICollection<GetResultIdsResponse.Types.MapTaskResult> GetResultIds(IEnume
   {
     using var _ = Logger?.LogFunction(taskId);
 
-    var resultId = GetResultIds(new[]
-                                {
-                                  taskId,
-                                })
-                   .Single()
-                   .ResultIds.Single();
+    var resultId = Retry.WhileException(5,
```
Comment (aneojgurhem) [inline Client/src/Common/Submitter/BaseClientSubmitter.cs:546]: wasn't that done in another pr ?

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Common/TaskPayload.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ArmoniK.EndToEndTests.Common
+{
+  public class TaskPayload
+  {
+    public int[] Vector           { get; set; }
+    public int      WorkloadTimeInMs { get; set; }
```
Comment (aneojgurhem) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Common/TaskPayload.cs:34]: formatting

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/CheckTypeOfSubmission/CheckAllSubmissionsClientTest.cs [hunk 6]
```diff
@@ -107,7 +107,7 @@ public void Cleanup()
     IEnumerable<string> taskIds;
     if (submissionType == SubmissionType.Sequential)
     {
-      taskIds = listOfPayload.Select(symphonyTestHelper_.SessionService.SubmitTask)
+      taskIds = listOfPayload.Select(x => symphonyTestHelper_.SessionService.SubmitTask(x))
```
Comment (aneojgurhem) [inline Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/CheckTypeOfSubmission/CheckAllSubmissionsClientTest.cs:110]: I'm not sure that should change

File: Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/LargeSubmitAsync/LargeSubmitAsyncClient.cs (hunk 3 above)
Comment (dbrasseur-aneo) [reply Tests/ArmoniK.EndToEndTests/ArmoniK.EndToEndTests.Client/Tests/LargeSubmitAsync/LargeSubmitAsyncClient.cs]: File doesn't exist anymore

File: Client/src/Unified/Services/Submitter/Service.cs [hunk 7]
```diff
@@ -146,78 +144,91 @@ public class Service : AbstractClientService, ISubmitterService
...
+                                      {
+                                        var taskIds =
+                                          SessionService.SubmitTasksWithDependencies(groupBlockRequest.Select(x => new
+                                                                                                                Tuple<string, byte[],
+                                                                                                                  IList<string>>(x.ResultId.ToString(),
+                                                                                                                                 x.Payload!.Serialize(),
+                                                                                                                                 null)),
+                                                                                     1,
+                                                                                     groupBlockRequest.First()
+                                                                                                      .TaskOptions);
```
[the remark this answers is not in the document]
Comment (dbrasseur-aneo) [reply Client/src/Unified/Services/Submitter/Service.cs:173]: The .Equals of TaskOptions looks pretty good for that purpose

[the remark this answers is not in the document]
Comment (aneojgurhem) [reply Client/src/Unified/Services/Submitter/Service.cs:173]: does it look into the TaskOptions.Options ?

[the remark this answers is not in the document]
Comment (dbrasseur-aneo) [reply Client/src/Unified/Services/Submitter/Service.cs:173]: Yes it does
